File: app/src/main/java/gesac/com/pickitem/ItemActivity.java
package gesac.com.pickitem;

import android.app.Activity;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.text.InputType;
import android.view.KeyEvent;

import com.google.gson.Gson;

import gesac.com.BR;
import gesac.com.R;
import gesac.com.databinding.ActivityItemBinding;
import gesac.com.scanbag.model.Journal;
import gesac.com.uitity.CodeUtil;
import gesac.com.uitity.WarnSPlayer;

public class ItemActivity extends Activity {
    ActivityItemBinding binding;

    Journal journal;
    ItemAdapter adapter;

    private int codetype = -1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initView();
    }

    private void initView() {
        binding = DataBindingUtil.setContentView(this, R.layout.activity_item);
        Intent it = getIntent();
        journal = new Gson().fromJson(it.getStringExtra("jour"), Journal.class);

        binding.itemstrEt.setInputType(InputType.TYPE_NULL);
        binding.jouridTv.setText(journal.getJourid());
        adapter = new ItemAdapter(this, BR.item, journal.getItemlist());
        binding.setItemadapter(adapter);
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_F8) {
            binding.itemstrEt.requestFocus();
            binding.itemstrEt.selectAll();
        }
        return super.onKeyDown(keyCode, event);
    }

    @Override
    public boolean onKeyUp(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_F8) {
            // 1、获取二维码2、拆分字符串3、匹配物料编号
            String strcode = binding.itemstrEt.getText().toString();
            binding.itemstrEt.setText("");
            adapter.setiBag(CodeUtil.subCode(strcode));
            codetype = CodeUtil.whichType(strcode);
            adapter.setType(codetype);
            //TODO iBag与list比较
            if (adapter.getiBag() != null) {
                int position = CodeUtil.isInItems(codetype, adapter.getiBag(), journal.getItemlist());
                if (position != -1) {
                    //TODO 若匹配则按钮可按
                    WarnSPlayer.playsound(this, R.raw.matchscd);
                    adapter.setIn(position);
                    binding.itemlist.setSelection(position - 1);
                } else {
                    WarnSPlayer.playsound(this, R.raw.error);
                }
            }
        }
        return super.onKeyUp(keyCode, event);
    }
}

File: app/src/main/java/gesac/com/pickitem/prod/ItemActivity.kt
package gesac.com.pickitem.prod

import android.os.Bundle
import android.text.InputType
import android.util.Log
import android.view.KeyEvent
import com.google.gson.Gson
import gesac.com.R
import gesac.com.databinding.ActivityItemBinding
import gesac.com.pickitem.model.Journal
import gesac.com.uitity.BaseActivity
import gesac.com.uitity.CodeUtil
import gesac.com.uitity.WarnSPlayer

class ItemActivity : BaseActivity<ActivityItemBinding>() {
    private lateinit var journal: Journal
    private lateinit var adapter: ItemAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initialize(R.layout.activity_item)
    }

    override fun initialize(LayoutId: Int) {
        super.initialize(LayoutId)
        val it = intent //获取得到的日记账数据Intent
        journal = Gson().fromJson(it.getStringExtra("jour"), Journal::class.java)
        Log.d(tag, journal.itemlist.size.toString())
        Log.d(tag, journal.jourid)

        bind.jouridTv.text = journal.jourid
        adapter = ItemAdapter(this, journal.itemlist)
        bind.itemlist.adapter = adapter
        bind.itemstrEt.inputType = InputType.TYPE_NULL
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        if (keyCode == KeyEvent.KEYCODE_F8) {
            bind.itemstrEt.requestFocus()
            bind.itemstrEt.selectAll()
        }
        return super.onKeyDown(keyCode, event)
    }

    override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
        if (keyCode == KeyEvent.KEYCODE_F8) {
            val strcode = bind.itemstrEt.text.toString()
            bind.itemstrEt.setText("")
            //转化扫描到的二维码
            val iBag = CodeUtil.subCode(strcode)
            adapter.iBag = iBag
            //判断二维码的类型
            val codetype = CodeUtil.whichType(strcode)
            adapter.type = codetype
            if (adapter.iBag != null) {//判断物料是否在该日记账内，在则打印按钮可按
                val position = CodeUtil.isInItems(codetype, adapter.iBag, journal.itemlist)
                if (position != -1) {
                    WarnSPlayer.playsound(this, R.raw.matchscd)
                    adapter.setIn(position)
                    bind.itemlist.setSelection(position - 1)
                } else {
                    WarnSPlayer.playsound(this, R.raw.error)
                }
            }
        }
        return super.onKeyUp(keyCode, event)
    }
}
